src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3949,19 +3950,54 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
 
                 MemberResolutionResult<MethodSymbol> memberResolutionResult;
                 ImmutableArray<MethodSymbol> candidateConstructors;
-                if (TryPerformConstructorOverloadResolution(
-                    initializerType,
-                    analyzedArguments,
-                    WellKnownMemberNames.InstanceConstructorName,
-                    errorLocation,
-                    false, // Don't suppress result diagnostics
-                    diagnostics,
-                    out memberResolutionResult,
-                    out candidateConstructors,
-                    allowProtectedConstructorsOfBaseType: true))
+                bool found;
+                MethodSymbol resultMember;
+                ImmutableArray<int> argsToParamsOpt;
+                bool isExpanded;
+                if (constructor is SynthesizedRecordCopyCtor)
+                {
+                    resultMember = SynthesizedRecordCopyCtor.FindCopyConstructor(baseType);
```

Comment:
>resultMember [](start = 20, length = 12)

Should report use-site diagnostics for found constructor

Reply:
Done, but didn't yet find scenario that could hit this. Let me know if I should file follow-up issue or add note to test plan.

---
In reply to: [438762631](https://github.com/dotnet/roslyn/pull/45022#discussion_r438762631) [](ancestors = 438762631)